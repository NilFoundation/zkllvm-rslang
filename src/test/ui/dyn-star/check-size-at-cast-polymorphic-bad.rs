#![feature(dyn_star)]
#![allow(incomplete_features)]

use std::fmt::Debug;

fn dyn_debug(_: (dyn* Debug + '_)) {

}

fn polymorphic<T: Debug + ?Sized>(t: &T) {
    dyn_debug(t);
    //~^ ERROR `&T` needs to be a pointer-sized type
}

fn main() {}
