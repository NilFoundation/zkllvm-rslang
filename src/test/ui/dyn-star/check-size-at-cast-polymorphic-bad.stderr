error[E0277]: `&T` needs to be a pointer-sized type
  --> $DIR/check-size-at-cast-polymorphic-bad.rs:11:15
   |
LL |     dyn_debug(t);
   |               ^ `&T` needs to be a pointer-sized type
   |
   = help: the trait `PointerSized` is not implemented for `&T`
help: consider introducing a `where` clause, but there might be an alternative better way to express this requirement
   |
LL | fn polymorphic<T: Debug + ?Sized>(t: &T) where &T: PointerSized {
   |                                          ++++++++++++++++++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
